frontend/src/employee-frontend/components/messages/MessagesPageContext.tsx

```diff
@@ -103,10 +110,28 @@ export const MessagesPageContextProvider = React.memo(
   }) {
     const [selectedAccount, setSelectedAccount] = useState<AccountView>()
 
-    const [accounts, setAccounts] = useState<Result<MessageAccount[]>>(
-      Loading.of()
-    )
-    const loadAccounts = useRestApi(getMessagingAccounts, setAccounts)
+    const [selectedUnit, setSelectedUnit] = useState<
+      SelectOptionProps | undefined
+    >()
+
+    const [accountsRequest, setAccountsRequest] = useState<
+      Result<MessageAccount[]>
+    >(Loading.of())
+    const loadAccounts = useRestApi(getMessagingAccounts, setAccountsRequest)
+    const [accounts, setAccounts] = useState<MessageAccount[]>()
```

Comment:
I don't really like the naming here as it doesn't indicate that the result set is filtered.

Reply:
Moreover, the whole filtering could be done in the component where it's needed. You could just `.map()` the original `accounts: Result<MessageAccount[]>` and filter from there.